bench/fixture.ts

```diff
@@ -57,10 +68,46 @@ export class TokenManager {
     await token.mint(settlement.address, LOTS);
 
     for (const trader of this.traders) {
-      await token.mint(trader.address, LOTS);
+      await token.mint(trader.address, LOTS.mul(3));
       await token.connect(trader).approve(vaultRelayer.address, LOTS);
+      await token.connect(trader).approve(vault.address, LOTS.mul(2));
+      await vault.connect(trader).depositToInternalBalance([
+        {
+          token: token.address,
+          amount: LOTS,
+          sender: trader.address,
+          recipient: trader.address,
+        },
+      ]);
     }
 
+    await balancerPool.registerTokens(
+      [token.address],
+      [ethers.constants.AddressZero],
+    );
+    await token.mint(pooler.address, LOTS);
+    await token.connect(pooler).approve(vault.address, LOTS);
+    const existingTokens = this.instances.map(({ address }) => address);
+    const zeros = this.instances.map(() => 0);
+    await vault.connect(pooler).joinPool(
+      await balancerPool.getPoolId(),
+      pooler.address,
+      pooler.address,
+      [...existingTokens, token.address],
+      [...zeros, LOTS],
+      false,
+      // NOTE: The mock pool uses this for encoding the pool share amounts
+      // that a user (here `pooler`) gets when joining the pool (first value)
+      // as well as the pool fees (second value).
+      ethers.utils.defaultAbiCoder.encode(
+        ["uint256[]", "uint256[]"],
+        [
+          [...zeros, LOTS],
+          [...zeros, 0],
```

Comment:
Do we need to set this to a non-zero value to simulate an already used pool and avoid the first benchmark to cost more?

Reply:
This code gets called when adding a new token to the fixture. This means that all existing tokens went through this and already have non-zero pool balances.

Reply:
Right for pool balances, however it feels like the second array (I assume this is fees) is 0 and might be set to non-0 on the first trade.

Reply:
Changing this to a non-zero value had no effect on the benchmarks.